File: src/main/java/com/akkaspring/GreetingActor.java
package com.akkaspring;

import akka.actor.AbstractActor;
import org.springframework.beans.factory.config.ConfigurableBeanFactory;
import org.springframework.context.annotation.Scope;
import org.springframework.stereotype.Component;

@Component
@Scope(ConfigurableBeanFactory.SCOPE_PROTOTYPE)
public class GreetingActor extends AbstractActor {

    private GreetingService greetingService;

    public GreetingActor(GreetingService greetingService) {
        this.greetingService = greetingService;
    }

    public void onGreet(Greet message) {
        String name = message.getName();
        String greet = greetingService.greet(name);
        getSender().tell(greet, getSelf());
    }

    @Override
    public Receive createReceive() {
        return receiveBuilder()
                .match(Greet.class, this::onGreet)
                .build();
    }

    public static class Greet {

        private String name;

        public Greet(String name) {
            this.name = name;
        }

        public String getName() {
            return name;
        }

    }

}

File: src/main/kotlin/com/akkaspring/GreetingActor.kt
package com.akkaspring

import akka.actor.AbstractActor
import org.springframework.beans.factory.config.ConfigurableBeanFactory
import org.springframework.context.annotation.Scope
import org.springframework.stereotype.Component

@Component
class GreetingService {
    fun greet(name: String) = "Hello, $name"
}

@Component
@Scope(ConfigurableBeanFactory.SCOPE_PROTOTYPE)
class GreetingActor2(private val greetingService: GreetingService) : AbstractActor() {

    data class Greet(val name: String)

    fun onGreet(message: Greet) {
        greetingService.greet(message.name).let {
            sender.tell(it, self)
        }
    }

    override fun createReceive(): Receive {
        return receiveBuilder()
                .match(Greet::class.java, this::onGreet)
                .matchAny { msg ->
                    unhandled(msg)
                }
                .build()
    }
}